sn_node_manager/src/cmd/node.rs

```diff
@@ -492,6 +492,138 @@ pub async fn upgrade(
     Ok(())
 }
 
+/// This functions makes sure that we run exactly the provided count nodes at any time by stopping nodes or
+/// by adding + starting nodes if required.
+///
+/// The arguments here are used during add() mostly.
+pub async fn maintain_n_running_nodes(
+    max_nodes_to_run: u16,
+    data_dir_path: Option<PathBuf>,
+    env_variables: Option<Vec<(String, String)>>,
+    home_network: bool,
+    local: bool,
+    log_dir_path: Option<PathBuf>,
+    log_format: Option<LogFormat>,
+    metrics_port: Option<PortRange>,
+    node_port: Option<PortRange>,
+    owner: Option<String>,
+    peers: PeersArgs,
+    rpc_address: Option<Ipv4Addr>,
+    rpc_port: Option<PortRange>,
+    src_path: Option<PathBuf>,
+    url: Option<String>,
+    upnp: bool,
+    user: Option<String>,
+    version: Option<String>,
+    verbosity: VerbosityLevel,
+    start_node_interval: u64,
+) -> Result<()> {
```

Comment:
I am suprised this doesn't trigger a clippy complain that `too many arguments to be passed` 
maybe make those for `add` as a separate struct ?

Reply:
I think there is a global `allow()` macro. Most of the functions there has got like 10+ args. Might be out of scope for this PR. :/